src/Domain/DonationRepository.php

```diff
@@ -297,50 +280,36 @@ public function releaseMatchFunds(Donation $donation): void
-                            $this->logInfo("Released {$fundingWithdrawal->getAmount()} to funding {$funding->getId()}");
-                            $this->logInfo("New fund total for {$funding->getId()}: $newTotal");
-                        }
-
-                        return $totalAmountReleased;
+        $totalAmountReleased = '0.00';
+        try {
+            $lockStartTime = microtime(true);
+            $totalAmountReleased = $this->matchingAdapter->runTransactionally(
+                function () use ($donation, $totalAmountReleased) {
+                    foreach ($donation->getFundingWithdrawals() as $fundingWithdrawal) {
+                        $funding = $fundingWithdrawal->getCampaignFunding();
+                        $newTotal = $this->matchingAdapter->addAmount($funding, $fundingWithdrawal->getAmount());
+                        $totalAmountReleased = bcadd($totalAmountReleased, $fundingWithdrawal->getAmount(), 2);
+                        $this->logInfo("Released {$fundingWithdrawal->getAmount()} to funding {$funding->getId()}");
+                        $this->logInfo("New fund total for {$funding->getId()}: $newTotal");
                     }
-                );
-                $lockEndTime = microtime(true);
-                $releaseDone = true;
 
-                try {
-                    $this->removeAllFundingWithdrawalsForDonation($donation);
-                } catch (DBALException $exception) {
-                    $this->logError('Doctrine could not remove withdrawals after maximum tries');
+                    return $totalAmountReleased;
                 }
-            } catch (Matching\TerminalLockException $exception) {
-                $waitTime = round(microtime(true) - $lockStartTime, 6);
-                $this->logError(
-                    'Match release FINAL error: ID ' . $donation->getUuid() . ' got ' . get_class($exception) .
-                    " after {$waitTime}s on try #$releaseTries: {$exception->getMessage()}"
-                );
-                throw $exception; // Re-throw exception after logging the details if not recoverable
-            }
-        }
-
-        if (!$releaseDone) {
-            $this->logger->error(
-                "Match release FINAL error: ID {$donation->getUuid()} failed releasing after $releaseTries tries"
             );
+            $lockEndTime = microtime(true);
 
-            throw new DomainLockContentionException();
+            try {
+                $this->removeAllFundingWithdrawalsForDonation($donation);
+            } catch (DBALException $exception) {
+                $this->logError('Doctrine could not remove withdrawals after maximum tries');
+            }
+        } catch (Matching\TerminalLockException $exception) {
+            $waitTime = round(microtime(true) - $lockStartTime, 6);
+            $this->logError(
+                'Match release FINAL error: ID ' . $donation->getUuid() . ' got ' . get_class($exception) .
```

Comment:
```suggestion
                'Match release error: ID ' . $donation->getUuid() . ' got ' . get_class($exception) .
```

Reply:
Done in https://github.com/thebiggive/matchbot/pull/694/commits/dd02f5b231a6d35b368116d82753197d254fc4d3